frappe/public/js/frappe/model/perm.js

```diff
@@ -44,6 +44,7 @@ $.extend(frappe.perm, {
 	get_perm: (doctype, doc) => {
 		let perm = [{ read: 0, permlevel: 0 }];
 
+		// TODO: wrap in `frappe.model.with_doctype`, to avoid undefined meta?
```

Comment:
AFAIK, the doctype should be there when your Kanban view is loaded, it is fetched [here](https://github.com/frappe/frappe/blob/97429e801281e5c92d16e1d648040d5b299fc4d6/frappe/public/js/frappe/list/base_list.js#L165).

Reply:
I think the main reason why `with_doctype` isn't done here is that it's an async event, whereas the `perm.get_perm` and `perm.has_perm` methods are designed to be synchronous.